torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -3643,13 +3636,8 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::pointToPoint(
   // to wait() on the returned handle, so ProcessGroupNCCL can't know
   // when it's safe to release the input back to the allocator,
   // and the present call has no way to know it's not an isend.
-  // Therefore, we warn and fall back to the typical recordStream logic:
-  if (avoidRecordStreams_) {
-    TORCH_WARN_ONCE(
-        "TORCH_NCCL_AVOID_RECORD_STREAMS=1 has no effect for point-to-point "
-        "collectives.");
-  }
-
+  // Therefore, we warn and fall back to the typical recordStream logic.
+  // TODO( kwen2501 ): revisit this when we have a better solution.
```

Comment:
A few options here:
1. Change the requirement to be that the user has to call `.wait()` even on isend calls.
2. Add some logic to the destructor that does the `synchronizeStream` or something equivalent to block the destruction on the p2p op